src/plugins/vis_type_timelion/server/usage_collector/get_usage_collector.test.ts

```diff
@@ -0,0 +1,162 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getStats } from './get_usage_collector';
+import { createCollectorFetchContextMock } from 'src/plugins/usage_collection/server/mocks';
+
+const mockedSavedObjects = [
+  {
+    _id: 'visualization:timelion-123',
+    _source: {
+      type: 'visualization',
+      visualization: {
+        visState: JSON.stringify({
+          type: 'timelion',
+          title: 'timelion visualization 1',
+          params: {
+            expression: '.es(index=my-test,metric=avg:scripted-test-field)',
+          },
+        }),
+      },
+      updated_at: new Date().toUTCString(),
+    },
+  },
+  {
+    _id: 'visualization:timelion-321',
+    _source: {
+      type: 'visualization',
+      visualization: {
+        visState: JSON.stringify({
+          type: 'timelion',
+          title: 'timelion visualization 1',
+          params: {
+            expression: '.es(index=my-test,metric=avg:scripted-test-field)',
+          },
+        }),
+      },
+      updated_at: '2020-10-17T00:00:00',
+    },
+  },
+  {
+    _id: 'visualization:timelion-456',
+    _source: {
+      type: 'visualization',
+      visualization: {
+        visState: JSON.stringify({
+          type: 'timelion',
+          title: 'timelion visualization 2',
+          params: {
+            expression: '.es(index=my-test,metric=avg:test-field)',
+          },
+        }),
+      },
+      updated_at: new Date().toUTCString(),
+    },
+  },
+];
+
+const getMockCollectorFetchContext = (hits?: unknown[]) => {
+  const fetchParamsMock = createCollectorFetchContextMock();
+
+  fetchParamsMock.esClient.search = jest.fn().mockResolvedValue({ body: { hits: { hits } } });
+  fetchParamsMock.soClient.find = jest.fn().mockResolvedValue({
+    saved_objects: [
+      {
+        attributes: {
+          title: 'my-test',
+          fields: JSON.stringify([{ scripted: true, name: 'scripted-test-field' }]),
+        },
+      },
+    ],
+  });
+  return fetchParamsMock;
+};
+
+describe('Timelion visualization usage collector', () => {
+  const mockIndex = 'mock_index';
+
+  test('Returns undefined when no results found (undefined)', async () => {
+    const result = await getStats(
+      getMockCollectorFetchContext().esClient,
+      getMockCollectorFetchContext().soClient,
+      mockIndex
+    );
+
+    expect(result).toBeUndefined();
+  });
+
+  test('Returns undefined when no results found (0 results)', async () => {
+    const result = await getStats(
+      getMockCollectorFetchContext([]).esClient,
+      getMockCollectorFetchContext().soClient,
+      mockIndex
+    );
+
+    expect(result).toBeUndefined();
+  });
+
+  test('Returns undefined when no vega saved objects found', async () => {
```

Comment:
Why vega here?

Reply:
Sorry, some tests copy from vega, fix it